blivet/partitioning.py

```diff
@@ -916,7 +918,16 @@ def allocate_partitions(storage, disks, partitions, freespace):
         if part_type == parted.PARTITION_EXTENDED and \
            part_type != _part.req_part_type:
             log.debug("creating extended partition")
-            add_partition(disklabel, free, part_type, None)
+            ext = add_partition(disklabel, free, part_type, None)
+
+            # extedned partition took all free space -- make the logical partition 1 MiB smaller
+            if aligned_size > (ext.geometry.length) * LINUX_SECTOR_SIZE - EXTENDED_METADATA_SIZE:
```

Comment:
This should also be using the alignment grain size, since that is what we use when allocating the extended partition (except when the user specified the start sector for it).